src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -357,6 +359,9 @@ def _disable_aladdin_service(self):
         if self.cli_ctx.__class__.__name__ == 'DummyCli':
             return True
 
+        if is_autocomplete():
+            return True
```

Comment:
This can be retrieved from `self.cli_ctx.data['completer_active']`.

Reply:
Nice catch, I'll update #24765 too.